test/test-buildURL.js

```diff
@@ -294,4 +314,61 @@ describe('URL Builder:', function describeSuite() {
             assert.equal(expectation, result);
         });
     });
-});
\ No newline at end of file
+
+    describe('Calling buildURL()', function describeSuite() {
+        var client,
+            path = 'images/1.png';
+
+        beforeEach(function setupClient() {
+            client = new ImgixClient({
+            domain: 'test.imgix.net',
+            includeLibraryParam: false
+            });
+        });
```

Comment:
No change needed, but I prefer to use `beforeEach` (and other lifecycle methods) strictly for things like setting up DB connections and things. The key thing is that imo, they are best used for things that aren't "coupled" with the test. The client in this case, however, is coupled with the test, since for example the domain is being checked in the "idempotent operation" test. Therefore, I like to instead write these "setup" calls as just functions, and call them in the tests. 

For example, if there was a function like such:

```js
const setupClient = (config) => new ImgixClient({ domain: 'test.imgix.net', includeLibraryParam: false, ...config });
```

then the first test could be:

```js
it('is an idempotent operation with empty args', function testSpec() {
  const client = setupClient();
  const result1 = client.buildURL('', {});
  const result2 = client.buildURL('', {});
  assert.equal(result1, result2);
});
```

The main benefit for me is that the test is more isolated to me, and when reading the test, everything that is relevant to that test (apart from DB connection stuff, etc) is written in that test, and I don't have to go back and forth between different places in the code 😄 

An additional benefit this has is that things like the config can be modified on a test-by-test basis. Otherwise in some tests the client invariably ends up being created _again_, resulting in more code duplication, rather than less 😅

Reply:
Cool! Yeah, tbh I wasn't quite sure about the purpose of `beforeEach`, but tried to extrapolate a bit from the usage in this repo. This is good to know, I'll continue to think about this one.